cupy/_core/_reduction.pyx

```diff
@@ -59,7 +59,7 @@ cpdef function.Function _create_reduction_function(
 ${type_preamble}
 ${preamble}
 #define REDUCE(a, b) (${reduce_expr})
-#define POST_MAP(a) (${post_map_expr})
+#define POST_MAP(a, b) (${post_map_expr})
```

Comment:
You can leave POST_MAP macro's argument to be only `a` (that is, `POST_MAP(a)`) as it is the result of the reduction phase. Instead, let's modify the expansion of the macro `(${post_map_expr})` to take statements instead of an expression. In C grammar, a parenthesis `()` can only contain an expression. For multi-line macro, we usually use `do { ... } while (0)` loop;

Reply:
Hi, @takagi! Thanks for the comments! I'll think from that perspective and update the PR :)